resolwe_bio/management/commands/generate_etc.py

```diff
@@ -38,20 +37,24 @@ def create_etc(gene_ids, path, file_name):
         """Generate an expression time course."""
         genes = {}
         with open(os.path.join(path, file_name), 'w') as f:
-            csvwriter = csv.writer(f, delimiter=str('\t'))
-            times = [0, 4, 8, 12, 16, 20, 24]
-            csvwriter.writerow(['Gene'] + times)
-            with gzopen(gene_ids) as gene_ids:
+            # NOTE: Default line terminator is '\r\n'
+            # NOTE: Python2's csv module doesn't accept a unicode string for delimeter
+            csvwriter = csv.writer(f, delimiter=str('\t'), lineterminator='\n')
+            times = (0, 4, 8, 12, 16, 20, 24)
+            csvwriter.writerow(('Gene', ) + times)
+            with gzip.open(gene_ids, mode='rt') as gene_ids:
```

Comment:
Hmm... since this is a generator function, where we control the input files, couldn't we just fix the input file to have the correct line ending?
Also, can you give me an example of a wrong output generated by this function?

BTW, on Python 3, [using `gzip.open` in *text* mode](https://docs.python.org/3/library/gzip.html#gzip.open) actually wraps the `GzipFile` object in a `io.TextIOWrapper` object which has universal newlines enabled by default.